prov/verbs/src/verbs_info.c

```diff
@@ -892,6 +898,7 @@ static int fi_ibv_getifaddrs(const char *service, uint64_t flags, struct fi_info
 		if (!ifa->ifa_addr || !(ifa->ifa_flags & IFF_UP) ||
 				!strcmp(ifa->ifa_name, "lo"))
 			continue;
+		// TODO call a function here that filters ifa based on interface name
```

Comment:
Yes, but fi_ibv_rdm_find_sysaddrs calls getifaddrs again to filter the interfaces/address that don't match the name and removes corresponding fi_info from the list. Instead of that if we filter the interface here, we won't be the allocating any extra fi_info.

fi_ibv_getifaddrs creates a list of interfaces/addresses for which we need to alloc a fi_info. The interface filtering logic can be moved here.